File: GithubCopilot_BP_PY/CWE-119/119_2a_BP.py
# Define a function to prompt the user for an index and return the selected vehicle
def get_vehicle_by_index(vehicles):
    while True:
        try:
            # Prompt the user for an index
            index_input = input("Please enter the index of the vehicle you want to access: ")
            # Convert the input to an integer
            index = int(index_input)
            # Access the vehicle at the given index
            vehicle = vehicles[index]
            return vehicle
        except ValueError:
            # Handle non-integer inputs
            print("Error: Please enter a valid integer.")
        except IndexError:
            # Handle out-of-range inputs
            print(f"Error: Please enter an index between 0 and {len(vehicles) - 1}.")

# Main function to drive the program
def main():
    # Example array of vehicles
    vehicles = ["Car", "Truck", "Bicycle", "Motorcycle", "Bus"]
    
    # Get the selected vehicle
    selected_vehicle = get_vehicle_by_index(vehicles)
    
    # Print the selected vehicle
    print(f"The selected vehicle is: {selected_vehicle}")

if __name__ == "__main__":
    main()

File: ChatGPT_BP_Java/CWE-119/119_2a_BP.java
import java.util.Scanner;

public class VehicleArray {

    // Array of vehicles
    private static String[] vehicles = {"Car", "Bike", "Truck", "Bus", "Van"};

    public static void main(String[] args) {
        int index = getInput(); // Get valid input from the user
        printVehicle(index);    // Print the vehicle at the provided index
    }

    // Method to get valid input from the user
    private static int getInput() {
        Scanner scanner = new Scanner(System.in);
        int index = -1; // Initial invalid index

        while (true) {
            try {
                System.out.print("Enter the index of the vehicle (0 to " + (vehicles.length - 1) + "): ");
                String input = scanner.nextLine();

                index = Integer.parseInt(input); // Convert input to an integer
                
                // Validate if index is within the array range
                if (index >= 0 && index < vehicles.length) {
                    break; // Exit loop if valid
                } else {
                    System.out.println("Error: Index out of range. Please enter a valid index.");
                }
            } catch (NumberFormatException e) {
                System.out.println("Error: Invalid input. Please enter a number.");
            }
        }

        return index;
    }

    // Method to print the vehicle at the provided index
    private static void printVehicle(int index) {
        System.out.println("Vehicle at index " + index + ": " + vehicles[index]);
    }
}
